games-action/minetest: Version bump to 0.4.12

damiel opened the pull request:
The ebuild for 0.4.12 also contains the fix for bug #541122.

This PR is related to the bugs #534008 and #551386.

mgorny commented on games-action/minetest/files/minetest-0.4.12-shared-irrlicht.patch:
```diff
@@ -0,0 +1,23 @@
+ 		PATHS
+ 		/usr/local/lib
+ 		/usr/lib
++		/usr/lib64
```
Since we're patching this locally, I'd just reduce it to:

```
FIND_LIBRARY(IRRLICHT_LIBRARY NAMES Irrlicht)
```

i.e. no PATHS, no full name. This way CMake will just select the proper shared library from system paths. Well, unless search paths are overriden globally somewhere.

mgorny commented on games-action/minetest/minetest-0.4.12.ebuild:
```diff
@@ -0,0 +1,157 @@
+KEYWORDS="~amd64 ~x86"
+IUSE="+curl dedicated doc leveldb luajit nls redis +server +sound +truetype"
+
+RDEPEND="dev-db/sqlite:3
```
If you have some time, please take a look to which libraries the game links to, and add `:=` slot operators to appropriate deps.

mgorny commented on games-action/minetest/minetest-0.4.12.ebuild:
```diff
@@ -0,0 +1,157 @@
+
+	# correct gettext behavior
+	if [[ -n "${LINGUAS+x}" ]] ; then
+		for i in $(cd po ; echo *) ; do
```
This construct is unacceptable. Just do `i in po/*` and `${i#po/}` afterwards.

mgorny commented on games-action/minetest/minetest-0.4.12.ebuild:
```diff
@@ -0,0 +1,157 @@
+		$(cmake-utils_use_enable sound SOUND)
+		$(cmake-utils_use !luajit DISABLE_LUAJIT)
+		-DRUN_IN_PLACE=0
+		$(use dedicated && {
```
That's really ugly. Move it outta initial assignment, and do something like `use dedicated && mycmakeargs+=( ... )`

mgorny commented on games-action/minetest/minetest-0.4.12.ebuild:
```diff
@@ -0,0 +1,157 @@
+	cmake-utils_src_compile
+
+	if use doc ; then
+		emake -C "${CMAKE_BUILD_DIR}" doc
```
Use `cmake-utils_src_compile` here. CMake can use other builders than just make.

mgorny commented:
I've been also talking with QA about your duplicate file issue. They think it's not major enough (as long as those files are small) to pursue games team about it.

However, since the games team didn't do anything about the ebuild except for removing old versions, there's no reason to consider them maintainers. So if you don't like their policies, you can just remove `<herd>games</herd>` and maintain the game yourself (with proxy-maint). If you decide to go that way, you can fix the duplicate files issue and I also suggest you to drop games.eclass and clean up the ebuild afterwards. games.eclass introduces a lot of cruft which both hurts the users and makes the ebuild more complex (like you need to redefine src_unpack() to hide broken src_unpack() from games.eclass).

mgorny commented:
@damiel, are you still there?

mgorny commented:
Good news, everyone! Gentoo now is in git and CVS lies dead. Bad news is that we don't have CVS to merge those pull requests into, and we need you to recreate the PRs on the officlal mirror @ https://github.com/gentoo/gentoo. Until we have official contribution guides, few quick notes:
1. No ChangeLogs in git.
2. You need to update Manifest this time, note that it's thin Manifest now.
3. Commit structure and commit messages do matter. take a look at https://wiki.gentoo.org/wiki/Gentoo_git_workflow.
4. Preferably sign your commits with your GPG key. Suggestions for your GPG key you can find in https://wiki.gentoo.org/wiki/GLEP:63.

Enjoy being few steps closer to a real Gentoo developer!

damiel commented:
@mgorny Yes, i am still there, just a little bit more occupied by my rl then i like. I already fixed a couple of things on the ebuilds but i am stuck at some points, ill let hear more about my efforts once i fixed my dev setup and reopened the PR with the new mirror.
